src/sleep.h

```diff
@@ -57,12 +57,17 @@ static inline void
 nanosleep_sleep(sendpacket_t *sp _U_, const struct timespec *nap,
         struct timespec *now,  bool flush _U_)
 {
-    #ifdef _POSIX_C_SOURCE >= 200112L
         struct timespec sleep_until;
         timeradd_timespec(now, nap, &sleep_until);
+    #ifdef _POSIX_C_SOURCE >= 200112L
         clock_nanosleep(CLOCK_REALTIME, TIMER_ABSTIME, &sleep_until, NULL);
     #else
-        nanosleep(nap, NULL);
+        while (!sp->abort) {
+            if (timescmp(now, &sleep_until, >=)){
+                break;
+            }
+            get_current_time(now);
+        }
```

Comment:
I'm a bit concerned about this one. Converting nanosleep to a busy loop might consume many resources. Am I miss something?

Reply:
I thought it is a good idea to use absolute sleep instead of relative sleep when an application needs to be suspended and activated in a periodic time, precise periodic activation cannot be achieved with a relative sleep() or nanosleep() function call.  --> https://linux.die.net/man/3/clock_nanosleep

Do you think this solution is less efficient in ensuring precise periodic suspension and activation? 

In case of nanosleep we do the following.

1. calculate now
2. calculate nap time (from pcap file)
3. nanosleep(nap)

When CPU time is taken away from the process right after (2) before (3)  then the application will not be activated exactly at now+nap. It will be activated at now+context_switch+nap

With busy loop it is easier to eliminate the effect of context switch since the time until the program should do nothing useful is an absolute time. What is your view on this?